v2/introspector.go

```diff
@@ -0,0 +1,68 @@
+package car
+
+import (
+	"bufio"
+	"fmt"
+	"github.com/ipfs/go-cid"
+	carv1 "github.com/ipld/go-car"
+	internalio "github.com/ipld/go-car/v2/internal/io"
+	"io"
+)
+
+// Introspection captures the result of an Introspect call.
+type Introspection struct {
+	// The version, either 1 or 2.
+	Version uint64
+	// The root CIDs if any.
+	Roots []cid.Cid
+	// Indicates whether the index is present.
+	HasIndex bool
```

Comment:
As per @willscott's comment above - maybe this isn't necessary and `IndexOffset==0` would be enough? Introducing the possibility of `IndexOffset=0 && HasIndex=true` could be opening awkward doors (I can hear @warpfork gently whispering _"cardinality my friends, it's all about cardinality ..."_).